### DISCUSSION TITLE
How long do files in the tmp dir survive?

### DISCUSSION THREAD

**Discussion by jordyvandomselaar** [vercel/vercel#4957] [Help] [answered]
I'm using a key/value storage system that writes to the tmp dir because the project dir itself is not writable in Vercel. How long do files survive in the tmp dir before they're cleaned up?

**Comment by styfle** [comment, marked as answer]
You should not rely on a time limit in the `/tmp` directory, the cleanup time is not deterministic.

Perhaps you should use something like Redis for storing session state instead.

**Reply by jordyvandomselaar** [reply]
Thanks for the reply! That makes sense =)

**Reply by agustif** [reply]
> You should not rely on a time limit in the `/tmp` directory, the cleanup time is not deterministic.
> 
> Perhaps you should use something like Redis for storing session state instead.

What are the limits / restrictions on /tmp folder? Is this in the docs?

Thanks
